src/data_provider/src/packages/appxWindowsWrapper.h

```diff
@@ -0,0 +1,454 @@
+         * @param[in] registry Registry object pointing to where the data is located.
+         *
+         * @return Return name application.
+         */
+        const std::string getName(const std::string& fullName, const Utils::Registry& registry)
+        {
+            std::string name;
+
+            /*
+             * Example of  fullName
+             *
+             * Microsoft.Windows.Search
+             * Microsoft.SkypeApp
+             */
+            const auto keyName { Utils::split(fullName, '.').back() }; // Will only use the last vector element.
+            constexpr auto INVALID_NAME_APP { "@{" };
+
+            try
+            {
+                for (const auto& folder : registry.enumerate())
+                {
+                    std::string value;
+                    const Utils::Registry nameReg(m_key, m_userId  + "\\" + APPLICATION_STORE_REGISTRY + "\\" + m_appName + "\\" + folder + "\\Capabilities");
+
+                    if (nameReg.string("ApplicationName", value))
+                    {
+                        name = value;
+                    }
+                }
+            }
+            catch (...)
+            {
+                name.clear();
+            }
+
+            const auto pos { name.find(INVALID_NAME_APP) };
+
+            /*
+             * If the name starts with "@{" string then we need to look for the name on the cache registry.
+             * The name obtained is the key for the app name.
+             */
+            if (pos != std::string::npos && pos == 0)
+            {
+                try
+                {
+                    name = searchNameFromCacheRegistry(keyName, name);
+                }
+                catch (...)
+                {
+                    name.clear();
+                }
+            }
+
+            return name;
+        }
+
+        /*
+         * @brief Get application vendor name.
+         *
+         * @param[in] registry Registry object that points to where the sub-registries with the names of the registry with the data are located.
+         *
+         * @return Returns the vendor name
+         */
+        const std::string getVendor(const Utils::Registry& registry)
+        {
+            std::string vendor;
+
+            try
+            {
+                for (const auto& folder : registry.enumerate())
+                {
+                    const Utils::Registry fileReg(m_key, m_userId + "\\" + APPLICATION_STORE_REGISTRY + "\\" + m_appName + "\\" + folder + FILE_ASSOCIATIONS_REGISTRY, KEY_READ | KEY_QUERY_VALUE);
+                    vendor = searchPublisher(fileReg);
+
+                    if (vendor.empty())
+                    {
+                        const Utils::Registry urlReg(m_key, m_userId + "\\"  + APPLICATION_STORE_REGISTRY + "\\" + m_appName + "\\" + folder + URL_ASSOCIATIONS_REGISTRY, KEY_READ | KEY_QUERY_VALUE);
+                        vendor = searchPublisher(urlReg);
+                    }
```

Comment:
Is valid to add a break to the for here?
```suggestion
                    }
                    else
                    {
                        break;
                    }
```

Reply:
Hi @Dwordcito

Yes, we avoid continuing the loop unnecessarily. Thank you very much for your comment